### ISSUE TITLE
False negative on VA_FORMAT_STRING_USES_NEWLINE

### ISSUE THREAD

**Issue by gloNelson** [spotbugs/spotbugs#1501]
Hi,
I found a false negative on VA_FORMAT_STRING_USES_NEWLINE when I used SpotBugs to check my project.
Here is a demo:
```java
void test(){
    String var = "hello world";
    String.format(var + " GitHub.\n"); // no warning (false negative)
}

void test01(){
    String.format(""hello world"" + " GitHub.\n"); // warning
}

```
It can be seen that the problem lies in `var + ""`.

**Comment by NateDawg97** [comment]
I'm working on this for a class project.  Forked SpotBugs into my own repo and am investigating.  Thank you!

**Comment by NateDawg97** [comment]
Is there a way to traverse the variables and string constants in String.format calls while visiting the method in FormatStringChecker.java??

**Comment by ThrawnCA** [comment]
At a guess, the detector probably doesn't try to examine the format string if it's not a literal.
